### ISSUE TITLE
warn  when type parameter is unused

### ISSUE THREAD

**Issue by gdkrmr** [julia-vscode/StaticLint.jl#49]
The following is valid julia code, but I guess it should warn that `S` is superfluous:
```
julia> f(x::T) where T where S = 3
```
```
julia> f(x::T) where T where S = S
```
gives a runtime error and should probably be marked as an error.

**Comment by ZacLN** [comment]
https://github.com/julia-vscode/StaticLint.jl/pull/51
